addons/mail/static/src/core/public_web/messaging_menu.js

```diff
@@ -0,0 +1,166 @@
+            return;
+        }
+        this.markAsRead(thread);
+    }
+
+    markAsRead(thread) {
+        if (thread.needactionMessages.length > 0) {
+            thread.markAllMessagesAsRead();
+        }
+        if (thread.model === "discuss.channel") {
+            thread.markAsRead();
+        }
+    }
+
+    navigate(direction) {
+        if (this.notificationItems.length === 0) {
+            return;
+        }
+        const activeOptionId = this.state.activeIndex !== null ? this.state.activeIndex : 0;
+        let targetId = undefined;
+        switch (direction) {
+            case "first":
+                targetId = 0;
+                break;
+            case "last":
+                targetId = this.notificationItems.length - 1;
+                break;
+            case "previous":
+                targetId = activeOptionId - 1;
+                if (targetId < 0) {
+                    this.navigate("last");
+                    return;
+                }
+                break;
+            case "next":
+                targetId = activeOptionId + 1;
+                if (targetId > this.notificationItems.length - 1) {
+                    this.navigate("first");
+                    return;
+                }
+                break;
+            default:
+                return;
+        }
+        this.state.activeIndex = targetId;
+        this.notificationItems[targetId]?.scrollIntoView({ block: "nearest" });
+    }
+
+    onKeydown(ev) {
+        if (!this.dropdown.isOpen) {
+            return;
+        }
+        const hotkey = getActiveHotkey(ev);
+        switch (hotkey) {
+            case "enter":
+                if (this.state.activeIndex === null) {
+                    return;
+                }
+                this.notificationItems[this.state.activeIndex].click();
+                break;
+            case "tab":
+                this.navigate(this.state.activeIndex === null ? "first" : "next");
+                break;
+            case "arrowup":
+                this.navigate(this.state.activeIndex === null ? "first" : "previous");
+                break;
+            case "arrowdown":
+                this.navigate(this.state.activeIndex === null ? "first" : "next");
+                break;
+            default:
+                return;
+        }
+        ev.preventDefault();
+        ev.stopPropagation();
+    }
+
+    get notificationItems() {
+        return this.notificationList.el?.children ?? [];
+    }
+
+    get threads() {
+        return this.store.menuThreads;
+    }
+
+    /**
+     * @type {{ id: string, icon: string, label: string }[]}
+     */
+    get tabs() {
+        return [
+            {
+                icon: "fa fa-user",
+                id: "chat",
+                label: _t("Chat"),
+            },
+            {
+                icon: "fa fa-users",
+                id: "channel",
+                label: _t("Channel"),
+            },
+        ];
+    }
```

Comment:
`"livechat"` and `"whatsapp"` channels can also be access in public page. These patches should also be in `public` bundle